nir/src/main/scala/scala/scalanative/nir/Arg.scala

```diff
@@ -0,0 +1,13 @@
+package scala.scalanative.nir
+
+case class Arg(ty: Type, passConvention: Option[PassConvention] = None)
+
+sealed abstract class PassConvention
+
+object PassConvention {
+
+  case class Byval(ty: Type) extends PassConvention
+
+  case class Sret(ty: Type) extends PassConvention
+
+}
```

Comment:
Those three blank lines above are not really necessary. See other NIR hierarchies for code style hint.